src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFinderScoring.cpp

```diff
@@ -230,10 +238,151 @@ namespace OpenMS
+      {
+        signal_noise_estimators_identification.push_back(snptr);
+        native_ids_identification.push_back(transition_group_identification.getTransitions()[i].getNativeID());
+      }
+    }
+
+
+    OpenSwath_Scores idscores;
+
+    if (native_ids_identification.size() > 0)
+    {
+      scorer.calculateChromatographicIdScores(idimrmfeature, native_ids_identification, native_ids_detection, signal_noise_estimators_identification, idscores);
+
+      std::stringstream id_transition_names;
+      for (size_t i = 0; i < native_ids_identification.size(); i++)
+      {
+        if (i != 0)
+        {
+          id_transition_names << ";";
+        }
+        id_transition_names << native_ids_identification[i];
+      }
+      idscores.id_transition_names = id_transition_names.str();
+
+      idscores.id_num_transitions = native_ids_identification.size();
+      idscores.elution_model_fit_score = emgscoring_.calcElutionFitScore(idmrmfeature, transition_group_identification);
+    }
+
+    bool swath_present = (swath_map->getNrSpectra() > 0);
+    if (swath_present && su_.use_dia_scores_ && native_ids_identification.size() > 0)
+    {
+      std::stringstream ind_isotope_correlation, ind_isotope_overlap, ind_massdev_score;
+      for (size_t i = 0; i < native_ids_identification.size(); i++)
+      {
+        OpenSwath_Scores tmp_scores;
+        std::vector<TransitionType> id_transitions;
+        id_transitions.push_back(transition_group_identification.getTransition(native_ids_identification[i]));
+
+        scorer.calculateDIAIdScores(idimrmfeature, id_transitions, swath_map, diascoring_, tmp_scores);
+
+        if (i != 0)
+        {
+          ind_isotope_correlation << ";";
+          ind_isotope_overlap << ";";
+          ind_massdev_score << ";";
+        }
+        ind_isotope_correlation << tmp_scores.isotope_correlation;
+        ind_isotope_overlap << tmp_scores.isotope_overlap;
+        ind_massdev_score << tmp_scores.massdev_score;
+      }
+      idscores.ind_isotope_correlation = ind_isotope_correlation.str();
+      idscores.ind_isotope_overlap = ind_isotope_overlap.str();
+      idscores.ind_massdev_score = ind_massdev_score.str();
+    }
+
+    delete idimrmfeature;
+
+    return idscores;
+  }
+
   void MRMFeatureFinderScoring::scorePeakgroups(MRMTransitionGroupType& transition_group,
                                                 TransformationDescription& trafo, OpenSwath::SpectrumAccessPtr swath_map,
                                                 FeatureMap& output)
   {
+    MRMTransitionGroupType transition_group_detection, transition_group_identification, transition_group_identification_decoy;
+    splitTransitionGroupsDetection_(transition_group, transition_group_detection);
+    if (su_.use_uis_scores)
```

Comment:
again, I am not sure whether this should only be done when uis scores are used. You could still imagine that a user provides a traml with id and detect transitions but forgets / disables the UIS scores. Still in this case you would want to only use the identification transitions, right? So I suggest to remove the check `if (su_.use_uis_scores)` as I dont see a case where you really actually want to skip the `splitTransitionGroupsIdentification_` step. Or am I wrong?